mitodl/mitxonline
# Add URL routing for catalog for courses/programs and department

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/3690 and https://github.com/mitodl/hq/issues/2516

### Description (What does it do?)

Removes the possibility of courses being shown that are not in the count from the API. We were checking this on the front end, but the API call we're using checks for this now, so I removed that bit from the filter function as it COULD have, in a race condition, created an infinite loading loop.

When you go to catalog/programs or catalog/courses it will pick the correct tab. When you add a department slug, that department will be selected. This PR also introduces a slug to the department field & migrates in order to standardize those slugs and ensure there is nothing wild in the future. It also adds to the management command.


### How can this be tested?
Create a department via management command or admin, should be able to. Should have a slug.
Go to /catalog/programs, should be programs tab. 
Go to /catalog/courses/department-name and should get there, same with programs.
Go to /catalog/fake and should get all departments, same with programs.
Should then be able to click around and not get jumped back. Should have correct counts and everything still.
